api/tasks.go

```diff
@@ -595,11 +595,18 @@ type DispatchPayloadConfig struct {
 	File string
 }
 
+type TaskLifecycle struct {
+	Hook       string        `mapstructure:"hook"`
+	BlockUntil string        `mapstructure:"block_until"`
+	Deadline   time.Duration `mapstructure:"deadline"`
```

Comment:
Should this maybe be `Timeout` rather than `Deadline` because it's a duration rather than a discrete expiry timestamp?

Reply:
We've called this as a deadline to be consistent with progress related stanzas (e.g. progress_deadline, healthy_deadlines) all of which are durations:

```
$ git grep -e 'Deadline[ *]*time' nomad/structs/ api/
api/deployments.go:     ProgressDeadline  time.Duration
api/jobs.go:    HealthyDeadline  *time.Duration `mapstructure:"healthy_deadline"`
api/jobs.go:    ProgressDeadline *time.Duration `mapstructure:"progress_deadline"`
api/nodes.go:   ForceDeadline time.Time
api/nodes.go:   Deadline time.Duration
api/tasks.go:   HealthyDeadline *time.Duration `mapstructure:"healthy_deadline"`
nomad/structs/structs.go:       Deadline time.Duration
nomad/structs/structs.go:       ForceDeadline time.Time
nomad/structs/structs.go:       HealthyDeadline time.Duration
nomad/structs/structs.go:       ProgressDeadline time.Duration
nomad/structs/structs.go:       HealthyDeadline time.Duration
nomad/structs/structs.go:       ProgressDeadline time.Duration
```

I think we started with timeout but in RFC discussions, we switched to deadline for consistency.